sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -416,6 +415,97 @@ private object CosmosChangeFeedConfig {
   }
 }
 
+private case class CosmosThroughputControlConfig(groupName: String,
+                                                 targetThroughput: Option[Int],
+                                                 targetThroughputThreshold: Option[Double],
+                                                 globalControlDatabase: String,
+                                                 globalControlContainer: String,
+                                                 globalControlRenewInterval: Option[Duration],
+                                                 globalControlExpireInterval: Option[Duration])
+
+private object CosmosThroughputControlConfig {
+    private val enableThroughputControlSupplier = CosmosConfigEntry[Boolean](
+        key = "spark.cosmos.enableThroughputControl",
```

Comment:
after this PR merges please raise another PR for updating the docs:
https://github.com/Azure/azure-sdk-for-java/blob/feature/cosmos/spark30/sdk/cosmos/azure-cosmos-spark_3-0_2-12/docs/configuration-reference.md